File: src/python/algorithms/greedy/min_initial_energy_to_finish_tasks.py
def minimumEffort(tasks):
    # Sort by (required - cost) descending
    tasks = sorted(tasks, key=lambda x: (x[1] - x[0]), reverse=True)
    energy = 0
    for cost, req in tasks:
        energy = max(energy + cost, req)
    return energy

def run_tests():
    tests = [
        ([[1,2],[2,4],[4,8]], 8),
        ([[1,3],[2,4],[10,11],[10,12],[8,9]], 32),
        ([[1,7],[2,8],[3,9],[4,10]], 19),
        ([[5,9],[2,8],[3,7],[1,5]], 17),
        ([[1,2],[2,2],[3,3]], 6),
    ]
    for i, (tasks, expected) in enumerate(tests, 1):
        result = minimumEffort([list(t) for t in tasks])
        print(f"Test {i}: Output={result}, Expected={expected} [{'PASS' if result==expected else 'FAIL'}]")

if __name__ == "__main__":
    run_tests()

File: src/javascript/algorithms/greedy/minInitialEnergyToFinishTasks.js
function minimumEffort(tasks) {
    tasks.sort((a, b) => (b[1] - b[0]) - (a[1] - a[0]));
    let energy = 0;
    for (const [cost, req] of tasks) {
        energy = Math.max(energy + cost, req);
    }
    return energy;
}

function runTests() {
    const tests = [
        { tasks: [[1,2],[2,4],[4,8]], expected: 8 },
        { tasks: [[1,3],[2,4],[10,11],[10,12],[8,9]], expected: 32 },
        { tasks: [[1,7],[2,8],[3,9],[4,10]], expected: 19 },
        { tasks: [[5,9],[2,8],[3,7],[1,5]], expected: 17 },
        { tasks: [[1,2],[2,2],[3,3]], expected: 6 },
    ];
    tests.forEach((test, i) => {
        const result = minimumEffort(test.tasks.map(t => t.slice()));
        console.log(`Test ${i+1}: Output=${result}, Expected=${test.expected} [${result===test.expected?'PASS':'FAIL'}]`);
    });
}

runTests();
